confgenerator/testdata/invalid/linux/logging-processor_parse_json_type_missing_time_format_required_with_time_key/golden_error

```diff
@@ -0,0 +1 @@
+"time_format" is required when "TimeKey" is set
```

Comment:
[Might not be directly related to this PR, and not a huge deal] But should `TimeKey` be `time_key` instead? It could be related to the common util: https://github.com/GoogleCloudPlatform/ops-agent/blob/11dc4f1cf20e99cbcd09c3c298b5845b4e7b0f59/confgenerator/config.go#L103

Ignore unless it's a low hanging fruit.

Reply:
After exploring the validator library, I've found this is a common request of getting the "Field" name (snake case) from the "Param" as discussed here : https://github.com/go-playground/validator/issues/597 . 

The other way is converting `TimeKey` to SnakeCase (`time_key`) using a go library, but i don't think that's a good solution. 

I think that an update to the https://github.com/go-playground/validator library would be the best solution. I will leave it as it is in this PR.